src/components/full-view-accommodation/full-view-accommodation.tsx

```diff
@@ -21,40 +33,90 @@ export interface ShowPhotosProps {
   id: string;
   open: boolean;
   onClose: () => void;
+  handleOpen: () => void;
+  isMobile?: boolean;
 }
 
-export default function ShowPhotos({ id, open, onClose }: ShowPhotosProps) {
-  const { data } = UseGetAllMediaQuery(id);
-  const [currentIndex, setCurrentIndex] = React.useState(0);
+export default function ShowPhotos({ id, open, onClose, handleOpen, isMobile }: ShowPhotosProps) {
+  const { data } = useGetAllMediaQuery(id, open);
 
-  const handleNext = () => {
-    setCurrentIndex((prevIndex) => Math.min(prevIndex + 1, data?.length ?? 0 - 1));
-  };
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const dialogRef = useRef<HTMLDivElement>(null);
+
+  const handleNext = useCallback(() => {
+    setCurrentIndex((prevIndex) => Math.min(prevIndex + 1, (data?.length ?? 0) - 1));
+  }, [data]);
 
-  const handlePrev = () => {
+  const handlePrev = useCallback(() => {
     setCurrentIndex((prevIndex) => Math.max(prevIndex - 1, 0));
-  };
+  }, []);
 
-  const handleClose = (event: React.MouseEvent<HTMLButtonElement>) => {
+  const handleClose = (event: MouseEvent<HTMLButtonElement>) => {
     event.stopPropagation();
     onClose();
   };
 
+  useEffect(() => {
+    const dialogElement = dialogRef.current as HTMLDivElement;
+
+    const handleKeyDown = (event: KeyboardEvent) => {
+      switch (event.key) {
+        case 'ArrowLeft':
+          handlePrev();
+          break;
+        case 'ArrowRight':
+          handleNext();
+          break;
+        default:
+          break;
+      }
+    };
+
+    dialogElement?.addEventListener('keydown', handleKeyDown as () => void);
```

Comment:
You don't need type casting here. Browser has a native type of Keyboard event and it will be compatible with addEventListener handler